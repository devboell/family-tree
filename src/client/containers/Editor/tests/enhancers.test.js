import { print } from 'graphql'

import {
    RELATIONS_QUERY,
    CREATE_PERSON_MUTATION,
    UPDATE_PERSON_MUTATION,
    REMOVE_PERSON_MUTATION,
    relationsQueryProps,
    refetchQueries,
    createPersonMutationProps,
    updatePersonMutationProps,
    removePersonMutationProps,
  } from '../enhancers'

describe('Editor enhancers', () => {
  it('RELATIONS_QUERY has a known shape', () => {
    expect(print(RELATIONS_QUERY)).toMatchSnapshot()
  })

  it('CREATE_PERSON_MUTATION has a known shape', () => {
    expect(print(CREATE_PERSON_MUTATION)).toMatchSnapshot()
  })

  it('UPDATE_PERSON_MUTATION has a known shape', () => {
    expect(print(UPDATE_PERSON_MUTATION)).toMatchSnapshot()
  })

  it('REMOVE_PERSON_MUTATION has a known shape', () => {
    expect(print(REMOVE_PERSON_MUTATION)).toMatchSnapshot()
  })

  it('correctly destructures query props', () => {
    const queryProps = {
      data: {
        relations: [],
        loading: true,
      },
    }
    expect(relationsQueryProps(queryProps)).toEqual(queryProps.data)
  })

  it('refetchQueries specifies correct query', () => {
    const expected = [
      {
        query: RELATIONS_QUERY,
      },
    ]
    expect(refetchQueries).toEqual(expected)
  })

  describe('mutation props', () => {
    const mutate = jest.fn()
    const values = { id: 1, name: 'bla' }

    beforeEach(() => {
      mutate.mockReset()
    })

    it('createPersonMutationProps', () => {
      const mutation = createPersonMutationProps({ mutate })
      mutation.createPersonMutation(values)
      expect(mutate).toHaveBeenCalledWith({ refetchQueries, variables: values })
    })

    it('updatePersonMutationProps', () => {
      const mutation = updatePersonMutationProps({ mutate })
      mutation.updatePersonMutation(values)
      expect(mutate).toHaveBeenCalledWith({ refetchQueries, variables: values })
    })

    it('removePersonMutationProps', () => {
      const mutation = removePersonMutationProps({ mutate })
      mutation.removePersonMutation(values)
      expect(mutate).toHaveBeenCalledWith({ refetchQueries, variables: values })
    })
  })
})
