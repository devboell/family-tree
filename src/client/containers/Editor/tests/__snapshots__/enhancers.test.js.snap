// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor enhancers CREATE_PERSON_MUTATION has a known shape 1`] = `
"mutation createPersonMutation($name: String, $title: String, $birthDate: Int, $deathDate: Int, $gender: String, $isMonarch: Boolean, $bornToId: ID, $partners: [PartnerInput]) {
  createPerson(name: $name, title: $title, birthDate: $birthDate, deathDate: $deathDate, gender: $gender, isMonarch: $isMonarch, bornToId: $bornToId, partners: $partners) {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;

exports[`Editor enhancers RELATIONS_QUERY has a known shape 1`] = `
"query relationsQuery {
  relations {
    id
    person1 {
      name
    }
    person2 {
      name
    }
  }
}
"
`;

exports[`Editor enhancers REMOVE_PERSON_MUTATION has a known shape 1`] = `
"mutation removePersonMutation($id: ID!) {
  removePerson(id: $id)
}
"
`;

exports[`Editor enhancers UPDATE_PERSON_MUTATION has a known shape 1`] = `
"mutation updatePersonMutation($id: ID!, $name: String, $title: String, $birthDate: Int, $deathDate: Int, $gender: String, $isMonarch: Boolean, $bornToId: ID, $partners: [PartnerInput]) {
  updatePerson(id: $id, name: $name, title: $title, birthDate: $birthDate, deathDate: $deathDate, gender: $gender, isMonarch: $isMonarch, bornToId: $bornToId, partners: $partners) {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;
